src/plugins/intel_gna/transformations/transpose_to_pwl.cpp

```diff
@@ -0,0 +1,425 @@
+    double err_pct = 0;
+    auto segments = pwl_search<T>(node, allowed_err_pct, err_pct);
+    if (segments.size() < 2) {
+        return false;
+    }
+
+    std::vector<double> m(segments.size() - 1);
+    std::vector<double> b(segments.size() - 1);
+    std::vector<double> alpha(segments.size());
+    for (size_t i = 0; i < segments.size() - 1; i++) {
+        m[i] = segments[i].m;
+        b[i] = segments[i].b;
+        alpha[i] = segments[i].alpha;
+    }
+    alpha[segments.size() - 1] = segments[segments.size() - 1].alpha;
+
+    auto m_constant = std::make_shared<ngraph::opset8::Constant>(ngraph::element::Type_t::f64,
+        ngraph::Shape{segments.size() - 1}, m);
+    auto b_constant = std::make_shared<ngraph::opset8::Constant>(ngraph::element::Type_t::f64,
+        ngraph::Shape{segments.size() - 1}, b);
+    auto alpha_constant = std::make_shared<ngraph::opset8::Constant>(ngraph::element::Type_t::f64,
+        ngraph::Shape{segments.size()}, alpha);
+    auto pwl = std::make_shared<Pwl>(node->input(0).get_source_output(), m_constant, b_constant, alpha_constant);
+    pwl->set_friendly_name(node->get_friendly_name());
+    ngraph::copy_runtime_info(node, pwl);
+    replace_node(node, pwl);
+    return true;
+}
+
+template<typename T>
+bool transpose_to_pwl(const std::tuple<T>& args,
+                      const std::shared_ptr<ngraph::Node>& node,
+                      double allowed_err_pct);
+
+template<typename T, typename ...Types>
+bool transpose_to_pwl(const std::tuple<T, Types...>& args,
+                      const std::shared_ptr<ngraph::Node>& node,
+                      double allowed_err_pct) {
+    auto op = std::dynamic_pointer_cast<T>(node);
+    if (op) {
+        return transpose_to_pwl(op, allowed_err_pct);
+    }
+    return transpose_to_pwl<Types...>(std::tuple<Types...>(), node, allowed_err_pct);
+}
+
+template<typename T>
+bool transpose_to_pwl(const std::tuple<T>& args,
+                      const std::shared_ptr<ngraph::Node>& node,
+                      double allowed_err_pct) {
+    auto op = std::dynamic_pointer_cast<T>(node);
+    if (op) {
+        return transpose_to_pwl(op, allowed_err_pct);
+    }
+    return false;
+}
+
+TransposeToPwl::TransposeToPwl(double allowed_err_pct) {
+    MATCHER_SCOPE(TransposeToPwl);
+    auto sigmoid = ngraph::pattern::wrap_type<ngraph::opset8::Sigmoid>({ ngraph::pattern::any_input() });
+    auto tanh = ngraph::pattern::wrap_type<ngraph::opset8::Tanh>({ ngraph::pattern::any_input() });
+    auto exp = ngraph::pattern::wrap_type<ngraph::opset8::Exp>({ ngraph::pattern::any_input() });
+    auto power = ngraph::pattern::wrap_type<ngraph::opset8::Power>({ ngraph::pattern::any_input(), ngraph::pattern::any_input() });
+    auto powerIE = ngraph::pattern::wrap_type<ngraph::op::PowerIE>({ ngraph::pattern::any_input() });
+    auto log = ngraph::pattern::wrap_type<ngraph::opset8::Log>({ ngraph::pattern::any_input() });
```

Comment:
There is a sense to add linear functions as well (ReLU). Maybe we can create a separate pass for them since algorithm is not required.